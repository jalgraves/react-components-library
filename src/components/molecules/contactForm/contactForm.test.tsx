import React from "react"
import renderer from "react-test-renderer"
import ContactForm from "./contactForm"

it('contact form snapshot', () => {
  const component = renderer.create(
    <ContactForm 
      ariaDetails="TestContactForm"
      buttonColor="red"
      buttonFontColor="white"
      buttonFontWeight="bold"
      buttonLetterSpacing=".3rem"
      contactReply="This is a reply"
      display="flex"
      flexFlow="column wrap"
      fontFamily="Arial, Helvetica, sans-serif"
      invalidEmailError="This is an invalid email error message"
      invalidPhoneError="This is an invalid phone number error message"
      labelFontWeight="bold"
      labelLetterSpacing=".3rem"
      pFontColor="yellow"
      pFontFamily="Arial, Helvetica, sans-serif"
      pFontWeight="900"
      pLetterSpacing=".2rem"
      pTextTransform="uppercase"
      requiredFieldError="This is an error message"
      width="100%"
    />
  )
  let tree = component.toJSON()! as any
  console.log("ContactForm")
  console.log(tree.props)
  console.log(tree.children)
  expect(tree).toMatchSnapshot()
})
