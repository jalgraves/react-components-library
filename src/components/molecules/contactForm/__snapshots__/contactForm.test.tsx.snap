// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact form snapshot 1`] = `
<div
  aria-details="TestContactForm"
  className="sc-guDMob bbhHqd"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <input
      name="name"
      onBlur={[Function]}
      onChange={[Function]}
    />
    <label>
      Email
    </label>
    <input
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
    />
    <label>
      Phone
    </label>
    <input
      name="phone"
      onBlur={[Function]}
      onChange={[Function]}
    />
    <label>
      Details
    </label>
    <textarea
      name="details"
      onBlur={[Function]}
      onChange={[Function]}
      rows={10}
    />
    <input
      type="submit"
    />
  </form>
</div>
`;
